ArmoniK.Extension.CSharp.DllWorker/LibraryLoader.cs

```diff
@@ -0,0 +1,404 @@
+
+        await File.WriteAllBytesAsync(zipFilePath,
+                                      libraryBytes,
+                                      cancellationToken);
+        /*
+        await using var fs = new FileStream(Path.Join(filePath,
+                                                      filename),
+                                            FileMode.Create,
+                                            FileAccess.ReadWrite,
+                                            FileShare.Read,
+                                            libraryBytes.Length);
+
+        fs.Seek(0,
+                SeekOrigin.Begin);
+        await fs.WriteAsync(libraryBytes,
+                            0,
+                            libraryBytes.Length,
+                            cancellationToken);
+        */
+      }
+      catch (Exception ex)
+      {
+        Logger.LogError("W");
+        throw new WorkerApiException(ex);
+      }
+
+      Logger.LogInformation("Extracting from archive {localZip}",
+                            Path.Join(filePath,
+                                      filename));
+
+      var extractedPath = ExtractArchive(filename,
+                                         filePath,
+                                         destinationPath,
+                                         pathToDllFile,
+                                         dllFileName);
+
+      var zipFile = Path.Join(filePath,
+                              filename);
+
+      File.Delete(zipFile);
+
+      Logger.LogInformation("Package {dynamicLibrary} successfully extracted from {localAssembly}",
+                            dynamicLibrary,
+                            extractedPath);
+
+      Logger.LogInformation($"Trying to load: {Path.Join(extractedPath,
+                                                         dllFileName)}");
+
+      var assembly = loadContext.LoadFromAssemblyPath(Path.Join(extractedPath,
+                                                                dllFileName));
+
+      var added = assemblyLoadContexts_.TryAdd(dynamicLibrary.ToString(),
+                                               (assembly, loadContext, Path.Join(extractedPath,
+                                                                                 dllFileName)));
+
+      if (!added)
+      {
+        throw new WorkerApiException($"Unable to add load context {dynamicLibrary}");
+      }
+
+      Logger.LogInformation($"Nb of current loaded assemblies: {assemblyLoadContexts_.Count}");
+
+      return dynamicLibrary.ToString();
+    }
+    catch (Exception ex)
+    {
+      Logger.LogError(ex.Message);
+      throw new WorkerApiException(ex);
+    }
+  }
+
+  public T GetClassInstance<T>(TaskLibraryDefinition dynamicLibrary)
+  {
+    try
+    {
+      assemblyLoadContexts_.TryGetValue(dynamicLibrary.ToString(),
+                                        out var assembly);
+      using (assembly.Item2.EnterContextualReflection())
```

Comment:
You could use a named tuple as value in the dictionary to avoid "Item1" and such